### PR TITLE
[receiver/sapm] - follow receiver contract

### PR DESCRIPTION
**Pull request by VihasMakwana**
**Description:** 
This is my first PR to ensure that all network receivers adhere to [the contract](https://github.com/open-telemetry/opentelemetry-collector/blob/df3c9e38a80ccc3b14705462be2e2e51c628a3b3/receiver/doc.go#L10) and maintain the correct order of operations. I also plan to submit additional PRs for other receivers in the future.

Follow receiver contract for `sapmreceiver`.
This also includes an internal `errorutil` package which will be used by other network receivers as well.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5909

**Testing:** Added

### CODE REVIEW

**Comment by VihasMakwana** [comment]
@atoulme this is good to go now! PTAL

File: internal/coreinternal/errorutil/http.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+package errorutil // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/errorutil"
+
+import (
+	"net/http"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+func HTTPError(w http.ResponseWriter, err error) {
```
Comment (atoulme) [inline internal/coreinternal/errorutil/http.go:12]: are you really going to reuse that code outside of sapmreceiver?

**Review by atoulme — APPROVED** [review]

File: internal/coreinternal/errorutil/http.go (hunk 1 above)
Comment (VihasMakwana) [reply internal/coreinternal/errorutil/http.go:12]: yes. look at this PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35327.

It's for loki receiver.

**Comment by VihasMakwana** [comment]
@atoulme merge conflicts are now resolved. I guess we can merge this.

**Review by mwear — APPROVED** [review]

**Pull request merged**